reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
+  assert(typeof P === 'string');
+  assert(Array.isArray(args));
```

Comment:
Wait, right, never mind about inconsistency. Cuz you mean all of them indirectly call InvokeOrNoop, right? Since PromiseInvokeOrNoop calls that now.